lldb/include/lldb/Core/PluginManager.h

```diff
@@ -487,6 +487,33 @@ class PluginManager {
 
   static LanguageSet GetAllTypeSystemSupportedLanguagesForExpressions();
 
+  // Scripted Interface
+  static bool
+  RegisterPlugin(llvm::StringRef name, llvm::StringRef description,
+                 ScriptedInterfaceCreateInstance create_callback,
+                 lldb::ScriptLanguage language,
+                 std::vector<llvm::StringRef> command_interpreter_usages,
+                 std::vector<llvm::StringRef> api_usages);
```

Comment:
Could the usages be their own objects? It would simplif the plugin API and if you add a dump method you could simplify the command object too.